akka-cluster/src/multi-jvm/scala/akka/cluster/SplitBrainQuarantineSpec.scala

```diff
@@ -0,0 +1,140 @@
+      pending
+    }
+
+    "join cluster" taggedAs LongRunningTest in {
+      awaitClusterUp(first, second, third, fourth)
+      enterBarrier("after-1")
+    }
+
+    "split brain" taggedAs LongRunningTest in {
+      runOn(first) {
+        testConductor.blackhole(first, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(first, fourth, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(second, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.blackhole(second, fourth, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("blackhole")
+      system.log.info("cluster split into [JVM-1, JVM-2] and [JVM-3, JVM-4] with blackhole")
+
+      within(15.seconds) {
+        runOn(first) {
+          cluster.down(third)
+          cluster.down(fourth)
+        }
+        runOn(first, second) {
+          awaitAssert {
+            cluster.state.members.collect { case m if m.status == MemberStatus.Up => m.address } should ===(
+              Set(address(first), address(second)))
+            cluster.state.members.size should ===(2)
+          }
+          system.log.info("JVM-3 and JVM-4 downed from JVM-1")
+        }
+
+        runOn(third) {
+          cluster.down(first)
+          cluster.down(second)
+        }
+        runOn(third, fourth) {
+          awaitAssert {
+            cluster.state.members.collect { case m if m.status == MemberStatus.Up => m.address } should ===(
+              Set(address(third), address(fourth)))
+            cluster.state.members.size should ===(2)
+          }
+          system.log.info("JVM-1 and JVM-2 downed from JVM-3")
+        }
+      }
+      enterBarrier("brain-split")
+
+      runOn(first) {
+        system.log.info("unblackholing cluster")
+        testConductor.passThrough(first, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(first, fourth, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(second, third, ThrottlerTransportAdapter.Direction.Both).await
+        testConductor.passThrough(second, fourth, ThrottlerTransportAdapter.Direction.Both).await
+      }
+      enterBarrier("unblackholed")
+
+      // must send some actual messages that would trigger Quarantine to be sure it does in fact not happen
+      runOn(first, second) {
+        system.actorSelection(RootActorPath(third) / "user").tell(Identify(None), ActorRef.noSender)
+        system.actorSelection(RootActorPath(fourth) / "user").tell(Identify(None), ActorRef.noSender)
+      }
+      runOn(third, fourth) {
+        system.actorSelection(RootActorPath(first) / "user").tell(Identify(None), ActorRef.noSender)
+        system.actorSelection(RootActorPath(second) / "user").tell(Identify(None), ActorRef.noSender)
+      }
+      Thread.sleep(3000)
+      runOn(first) {
```

Comment:
```suggestion
      runOn(first, second) {
```

To clarify, why are we sending `Identity` messages twice? 
Sending once from each node to each other node on the other side would be enough to get them all quarantined, no?

If we want/need to send twice, we probably want to include `second` to the party. 

Could you add a comment explaining why?

Reply:
It sends Identify to both nodes that was previously on the other side of the partition. It should send _from_ both nodes though, good catch.